octodns/provider/googlecloud.py

```diff
@@ -0,0 +1,363 @@
+#
+#
+#
+
+from __future__ import absolute_import, division, print_function, \
+    unicode_literals
+
+import re
+import shlex
+import time
+from logging import getLogger
+
+from google.cloud import dns
+
+from .base import BaseProvider
+from ..record import Record
+
+
+class _GoogleCloudRecordSetMaker(object):
```

Comment:
Fwiw, if this is modeled after the Route53Provider it's a little "extra" complicated b/c of the need to deal with geo records that combine/split into multiple rrsets. Looking at the code here I don't think that seems to be happening here so this may benefit from following the more straightforward providers, e.g. PowerDNS where there's a `_data_from_X` and `_params_for_X`. If so I think this case would just turn into something like `_rrset_for_A` methods on the provider itself. Feel free to lmk if I'm misreading something.

Reply:
I did use most of the others as role models for this one, mostly it was the Azure one. It's no particular reason, I guess it makes sense to move them all into the same class.